src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/InvokeWebRequestCommand.CoreClr.cs

```diff
@@ -65,7 +65,7 @@ internal override void ProcessResponse(HttpResponseMessage response)
 
                 WriteVerbose(string.Create(System.Globalization.CultureInfo.InvariantCulture, $"File Name: {Path.GetFileName(_qualifiedOutFile)}"));
 
-                StreamHelper.SaveStreamToFile(responseStream, outFilePath, this, response.Content.Headers.ContentLength.GetValueOrDefault(), perReadTimeout, _cancelToken.Token);
+                StreamHelper.SaveStreamToFile(responseStream, outFilePath, this, response.Content.Headers.ContentRange?.Length.GetValueOrDefault() ?? response.Content.Headers.ContentLength.GetValueOrDefault(), perReadTimeout, _cancelToken.Token);
```

Comment:
So now we are correctly showing how much left to download based on range size, not full file size? 
It would be interesting to compare with wget behavior. Does it show the remainder of the download or the full file size?

Reply:
Yes, range size instead of full size, but downloaded size is correct. Like my screenshots in issue #20206

Reply:
Since wget is the best known tool could you compare with its behavior? It would be nice to have identical behavior.

Reply:
I tested both curl and wget, but they have different behaviors.
curl displaying both downloaded size and range size of remaining part, also percentage and progress bar is all about remaining. just like, ignoring previously downloaded part. 
wget displaying both of full part. previously downloaded part plus current downloaded part, and complete full size, just like the behavior I changed to in pwsh.
![curl](https://github.com/PowerShell/PowerShell/assets/10334184/5e5600aa-2a2f-4b52-b674-a60e25f64b5d)
![wget](https://github.com/PowerShell/PowerShell/assets/10334184/b40f5825-e6a6-4e0e-8a93-38697357b11c)

Reply:
Thanks!
I hope WG will make a conclusion.
/cc @SteveL-MSFT  @TobiasPSP

Reply:
added to WG review queue

Reply:
@PowerShell/wg-powershell-cmdlets reviewed this.  We agreed that progress should show the total complete rather than the remaining complete for resuming a download.  Although we understand there is utility to showing progress for just the remaining, we believe the majority of users prefer the `wget` experience providing information on the total transfer than the remaining transfer.

Reply:
![image](https://github.com/PowerShell/PowerShell/assets/10334184/f6c0cba4-92b6-4094-850d-0e41ebabde37)

I have built and tested locally, confirmed that it shows total size and complete just like wget experience.

Reply:
Please add comment that for Resume scenario `response.Content.Headers.ContentRange?.Length` return full document size and `response.Content.Headers.ContentLength` return size of current range.

Reply:
anyone review this? @iSazonov